SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -214,6 +217,25 @@ TArray<Worker_ComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 		ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId()));
 	}
 
+	if (NetDriver->SpatialDebugger != nullptr)
+	{
+		if (SpatialSettings->bEnableUnrealLoadBalancer)
+		{
+			check(NetDriver->VirtualWorkerTranslator != nullptr);
+
+			VirtualWorkerId IntentVirtualWorkerId = NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId();
+
+			const PhysicalWorkerName* PhysicalWorkerName = NetDriver->VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(IntentVirtualWorkerId);
+			FColor InvalidServerTintColor = NetDriver->SpatialDebugger->InvalidServerTintColor;
+			FColor IntentColor = PhysicalWorkerName == nullptr ? InvalidServerTintColor : SpatialGDK::GetColorForWorkerName(*PhysicalWorkerName);
+
+			SpatialDebugging DebuggingInfo(SpatialConstants::INVALID_VIRTUAL_WORKER_ID, InvalidServerTintColor, IntentVirtualWorkerId, IntentColor, false);
+			ComponentDatas.Add(DebuggingInfo.CreateSpatialDebuggingData());
+		}
+
+		ComponentWriteAcl.Add(SpatialConstants::SPATIAL_DEBUGGING_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
```

Comment:
Do we want to have the SpatialDebugging SpatialOS component on entities if Unreal load balancing isn't enabled? I don't think any of the properties are useful for single-server? Given this, can we move this inside the above conditional and combine the two IFs into one conjunctive?

Reply:
As above, I think that we could put more information there in the future, so this is here for making adding those easier.

Reply:
Entity Id could be useful in single server I think